### ISSUE TITLE
"GLIBC_2.18 not found" error when using libmesos-bundle

### ISSUE THREAD

**Issue by tobilg** [mesosphere/dcos-commons#378]
The executors on current `master` seem to fail with the following error when speciying the libmesos-bundle:

```bash
Failed to load native Mesos library from /var/lib/mesos/slave/slaves/b019b00a-da04-4996-b6c0-35956ac1fbe6-S1/frameworks/b019b00a-da04-4996-b6c0-35956ac1fbe6-0000/executors/mongodb__fa1c51cb-9787-4eff-a2de-a4eadaf0c05d/runs/978d2b52-1ee7-4e90-8d86-51e5ef25841b/libmesos-bundle/lib/libmesos-1.0.1.so
Exception in thread "main" java.lang.UnsatisfiedLinkError: /var/lib/mesos/slave/slaves/b019b00a-da04-4996-b6c0-35956ac1fbe6-S1/frameworks/b019b00a-da04-4996-b6c0-35956ac1fbe6-0000/executors/mongodb__fa1c51cb-9787-4eff-a2de-a4eadaf0c05d/runs/978d2b52-1ee7-4e90-8d86-51e5ef25841b/libmesos-bundle/lib/libmesos-1.0.1.so: /lib64/libc.so.6: version `GLIBC_2.18' not found (required by /var/lib/mesos/slave/slaves/b019b00a-da04-4996-b6c0-35956ac1fbe6-S1/frameworks/b019b00a-da04-4996-b6c0-35956ac1fbe6-0000/executors/mongodb__fa1c51cb-9787-4eff-a2de-a4eadaf0c05d/runs/978d2b52-1ee7-4e90-8d86-51e5ef25841b/libmesos-bundle/lib/libstdc++.so.6)
	at java.lang.ClassLoader$NativeLibrary.load(Native Method)
	at java.lang.ClassLoader.loadLibrary0(ClassLoader.java:1941)
	at java.lang.ClassLoader.loadLibrary(ClassLoader.java:1824)
	at java.lang.Runtime.load0(Runtime.java:809)
	at java.lang.System.load(System.java:1086)
	at org.apache.mesos.MesosNativeLibrary.load(MesosNativeLibrary.java:159)
	at org.apache.mesos.MesosNativeLibrary.load(MesosNativeLibrary.java:188)
	at org.apache.mesos.MesosExecutorDriver.<clinit>(MesosExecutorDriver.java:52)
	at org.apache.mesos.executor.MesosExecutorDriverFactory.getDriver(MesosExecutorDriverFactory.java:14)
	at org.apache.mesos.executor.Main.run(Main.java:26)
	at org.apache.mesos.executor.Main.main(Main.java:19)
```

**Comment by tobilg** [comment]
I checked the glibc version of the CentOS 7 Docker image which is used for the dcos-docker-sdk Vagrant box:

```bash
$ cd tools/vagrant/ && vagrant ssh                                                                                                                        
Last login: Thu Dec 15 05:30:05 2016 from 10.0.2.2
[vagrant@dcos-docker-sdk ~]$ docker ps
CONTAINER ID        IMAGE                    COMMAND             CREATED             STATUS              PORTS               NAMES
6f130b4a64f7        mesosphere/dcos-docker   "/sbin/init"        3 weeks ago         Up 2 minutes                            dcos-docker-agent3
788b9159668b        mesosphere/dcos-docker   "/sbin/init"        3 weeks ago         Up 2 minutes                            dcos-docker-agent2
9a019d4196cc        mesosphere/dcos-docker   "/sbin/init"        3 weeks ago         Up 2 minutes                            dcos-docker-agent1
eab6bffa3553        mesosphere/dcos-docker   "/sbin/init"        3 weeks ago         Up 2 minutes                            dcos-docker-master1
[vagrant@dcos-docker-sdk ~]$ docker exec -it 6f bash
[root@dcos-docker-agent3 /]# rpm -q glibc
glibc-2.17-106.el7_2.8.x86_64
[root@dcos-docker-agent3 /]# 
```

So, it's version `2.17` but the libmesos-bundle seems to require `2.18`

**Comment by tobilg** [comment]
BTW, I get the same error when installing the `hello-world` framework...

**Comment by tobilg** [comment]
Any news on this issue?

**Comment by nickbp** [comment]
Should be fixed via #387 merged just now, maybe give that a shot?

**Comment by tobilg** [comment]
Works like a charm now, thanks!
